File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        L, R = 0, len(nums) - 1

        while L <= R:
            M = L + ((R - L) // 2)
            if (nums[M]==target):
                return M
            elif (nums[L]<=nums[M]):
                if (nums[L]<=target<nums[M]):
                    R = M - 1
                else:
                    L = M + 1
            elif (nums[M]<=nums[R]):
                if (nums[M]<target<=nums[R]):
                    L = M + 1
                else:
                    R = M - 1
        return -1

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int[] res = new int[nums.Length];
        for (int i=0; i<res.Length; i++)
        {
            int prod = 1;
            for (int j=0; j<res.Length; j++)
            {
                if (i!=j) { prod *= nums[j]; }
            }
            res[i] = prod;
        }
        return res;
    }
}
